src/routes/plans/[id]/+page.svelte

```diff
@@ -82,17 +82,32 @@
   import { removeQueryParam } from '../../../utilities/generic';
   import { isSaveEvent } from '../../../utilities/keyboardEvents';
   import { closeActiveModal, showPlanLockedModal } from '../../../utilities/modal';
+  import { featurePermissions } from '../../../utilities/permissions';
   import { Status } from '../../../utilities/status';
   import { getUnixEpochTime } from '../../../utilities/time';
   import type { PageData } from './$types';
 
   export let data: PageData;
 
   let compactNavMode = false;
+  let hasCreateViewPermission: boolean = false;
+  let hasUpdateViewPermission: boolean = false;
+  let hasExpandPermission: boolean = false;
+  let hasSchedulePermission: boolean = false;
+  let hasSimulatePermission: boolean = false;
+  let hasCheckConstraintsPermission: boolean = false;
   let planHasBeenLocked = false;
   let schedulingAnalysisStatus: Status | null;
   let windowWidth = 0;
 
+  $: hasCreateViewPermission = featurePermissions.view.canCreate(data.user);
+  $: hasUpdateViewPermission = $view !== null ? featurePermissions.view.canUpdate(data.user, $view) : false;
+  $: if ($plan) {
+    hasCheckConstraintsPermission = featurePermissions.constraints.canCheck(data.user, $plan);
+    hasExpandPermission = featurePermissions.expansionSequences.canExpand(data.user, $plan);
+    hasSchedulePermission = featurePermissions.schedulingGoals.canRun(data.user, $plan);
```

Comment:
This nav header button runs a scheduling analysis, not a full scheduling. So from my understanding, based on the roles google doc, you should be able to run an analysis as a user, without being a owner/collaborator?

![image](https://github.com/NASA-AMMOS/aerie-ui/assets/888818/1be402af-747d-4199-a34f-cb6f525d27a2)

There should be a separate `canAnalyze` featurePermissions method, that _I think_ handles this correctly.

Reply:
Ooh got it. Thanks! I've updated it to use the `canAnalyze` permission